catalog/spatial/registry/registry-admin-modules/registry-admin-local-ui/src/main/webapp/js/model/Association.js

```diff
@@ -142,15 +143,17 @@ define([
         },
         removeSegment: function (segment) {
```

Comment:
I know we generally prefer backbone-assocations, but since you're writing your own dynamic relationship handling logic here, it might be worth looking at backbone-relational, which is heavier but does all this stuff for you. I think you're writing/maintaining more logic than you need to be, and possibly using the wrong tool for the job.